llvm/lib/Transforms/Vectorize/VPlanHCFGBuilder.cpp

```diff
@@ -376,23 +311,21 @@ void PlainCFGBuilder::buildPlainCFG(
   for (BasicBlock *BB : RPO) {
     // Create or retrieve the VPBasicBlock for this BB.
     VPBasicBlock *VPBB = getOrCreateVPBB(BB);
-    VPRegionBlock *Region = VPBB->getParent();
     Loop *LoopForBB = LI->getLoopFor(BB);
     // Set VPBB predecessors in the same order as they are in the incoming BB.
     if (!isHeaderBB(BB, LoopForBB)) {
       setVPBBPredsFromBB(VPBB, BB);
```

Comment:
Can setVPBBPredsFromBB() also handle header blocks?

Reply:
For now, the code relies on the header having a specific predecessor order (preheader, then latch). We might be able to remove this restriction, because the dominance based approach could be generalized to handle it. It would require extra logic to determine which block is the preheader/latch.

Reply:
I just pushed an update to not rely on the predecessor order (see cc818012a9683fd4d1fe4dceb3a5e2fb32d91a44 for this change in isolation)